presto-main/src/main/java/io/prestosql/server/security/jwt/JwkDecoder.java

```diff
@@ -259,17 +283,55 @@ public String getFormat()
 
     public static class Keys
     {
-        private final List<Map<String, String>> keys;
+        private final List<Key> keys;
 
         @JsonCreator
-        public Keys(@JsonProperty("keys") List<Map<String, String>> keys)
+        public Keys(@JsonProperty("keys") List<Key> keys)
         {
             this.keys = ImmutableList.copyOf(requireNonNull(keys, "keys is null"));
         }
 
-        public List<Map<String, String>> getKeys()
+        public List<Key> getKeys()
         {
             return keys;
         }
     }
+
+    public static class Key
+    {
+        private final String kty;
+        private final Optional<String> kid;
+        private final Map<String, Object> other = new HashMap<>();
+
+        @JsonCreator
+        public Key(
+                @JsonProperty("kty") String kty,
+                @JsonProperty("kid") Optional<String> kid)
+        {
+            this.kty = requireNonNull(kty, "kty is null");
+            this.kid = requireNonNull(kid, "kid is null");
+        }
+
+        public String getKty()
+        {
+            return kty;
+        }
+
+        public Optional<String> getKid()
+        {
+            return kid;
+        }
+
+        @JsonAnyGetter
+        public Map<String, Object> getOther()
```

Comment:
It looks like every usage of this is of the form:
```
Object encodedY = key.getOther().get("y");
if (!(encodedY instanceof String)) {
    log.error("something about property being an invalid value");
    return Optional.empty();
}
if (isNullOrEmpty((String) encodedY)) {
    log.error("something the key not containing a required property");
    return Optional.empty();
}
```
I just added the extra detailed logging because it was easy.

I suggest we simplify this code to a method on the getter that tests if the value is a non empty string.  If it is not we return `Optional.empty()` otherwise we return the value.  Something like:
```
public Optional<String> getStringProperty(String name) {
    Object value = other.get(name);
    if (value instanceof String && !((String)value).isEmpty) {
        return Optional.of(value);
    }
    return Optional.empty();
}
```